app/Services/CampaignService.php

```diff
@@ -65,26 +65,34 @@ public function find($id)
     public function findAll($ids = [])
     {
         if ($ids) {
-            $campaigns = $this->cache->retrieveMany($ids);
+            $idsWithPrefix = [];
+
+            foreach ($ids as $key => $id) {
+                array_push($idsWithPrefix, $this->cache->setPrefix($id));
```

Comment:
This should all be encapsulated within the `CacheRepository` (so when you use `retrieveMany` or `storeMany` it automatically prepends the prefix to the given IDs).